fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1404,16 +1404,26 @@ ACTOR Future<GetReadVersionReply> getLiveCommittedVersion(ProxyCommitData* commi
 		g_traceBatch.addEvent("TransactionDebug", debugID.get().first(), "MasterProxyServer.getLiveCommittedVersion.confirmEpochLive");
 	}
 
-	vector<GetReadVersionReply> versions = wait(getAll(proxyVersions));
-	GetReadVersionReply rep;
-	rep.version = commitData->committedVersion.get();
+	state GetReadVersionReply rep;
 	rep.locked = commitData->locked;
 	rep.metadataVersion = commitData->metadataVersion;
 	rep.recentRequests = commitData->stats.getRecentRequests();
+	rep.version = commitData->committedVersion.get();
 
-	for (auto v : versions) {
-		if(v.version > rep.version) {
-			rep = v;
+	if (SERVER_KNOBS->ASK_READ_VERSION_FROM_MASTER) {
+		GetReadVersionReply liveCommittedVersionReply = wait(commitData->master.getLiveCommittedVersion.getReply(GetRawCommittedVersionRequest(debugID)));
+		if (liveCommittedVersionReply.version > rep.version) {
+			rep = liveCommittedVersionReply;
+		}
+	} else {
+		state vector<Future<GetReadVersionReply>> proxyVersions;
```

Comment:
In this case, previously the broadcast to all other proxies is run in parallel with `updateLastCommit()` or `commitData->lastCommitTime.whenAtLeast()`. And now after the change it would be run in sequential order. Since this is on GRV path, it's probably a good idea to retain that parallelism.

Reply:
Good catch, done.